contracts/modules/UniswapManager.sol

```diff
@@ -0,0 +1,159 @@
+            _wallet.invoke(pool, ethSwap, abi.encodeWithSignature("ethToTokenSwapInput(uint256,uint256)", 1, block.timestamp));
+            _wallet.invoke(_poolToken, 0, abi.encodeWithSignature("approve(address,uint256)", pool, tokenPool));
+            // add liquidity
+            _wallet.invoke(pool, ethPool, abi.encodeWithSignature("addLiquidity(uint256,uint256,uint256)",1, tokenPool, block.timestamp + 1));
+        }
+        else {
+            // swap some tokens for eth
+            (uint256 tokenSwap, uint256 tokenPool, uint256 ethPool) = computePooledValue(tokenPoolSize, ethPoolSize, _tokenAmount, _ethAmount, 0);
+            _wallet.invoke(_poolToken, 0, abi.encodeWithSignature("approve(address,uint256)", pool, tokenSwap + tokenPool));
+            _wallet.invoke(pool, 0, abi.encodeWithSignature("tokenToEthSwapInput(uint256,uint256,uint256)", tokenSwap, 1, block.timestamp));
+            // add liquidity
+            _wallet.invoke(pool, ethPool - 1, abi.encodeWithSignature("addLiquidity(uint256,uint256,uint256)",1, tokenPool, block.timestamp + 1));
+        }
+    }
+
+    function removeLiquidityFromUniswap(    
+        BaseWallet _wallet, 
+        address _poolToken, 
+        uint256 _amount
+    )
+        external 
+        onlyOwner(_wallet)
+        onlyWhenUnlocked(_wallet)        
+    {
+        address pool = uniswap.getExchange(_poolToken);
+        require(pool != address(0), "UM: The target token is not traded on Uniswap");
+        _wallet.invoke(pool, 0, abi.encodeWithSignature("removeLiquidity(uint256,uint256,uint256,uint256)",_amount, 1, 1, block.timestamp + 1));
+    }
+
+    /**
+     * @dev Computes the amount of tokens to swap and then pool given an amount of "major" and "minor" tokens,
+     * where there are more value of "major" tokens then "minor".
+     * @param _majorPoolSize The size of the pool in major tokens
+     * @param _minorPoolSize The size of the pool in minor tokens
+     * @param _majorAmount The amount of major token provided
+     * @param _minorAmount The amount of minor token provided
+     * @param _minorInMajor The amount of minor token converted to major (optional)
+     * @return the amount of major tokens to first swap and the amount of major and minor tokens that can be added to the pool after.
+     */
+    function computePooledValue(
+        uint256 _majorPoolSize,
+        uint256 _minorPoolSize, 
+        uint256 _majorAmount,
+        uint256 _minorAmount, 
+        uint256 _minorInMajor
+    ) 
+        internal 
+        view 
+        returns(uint256 _majorSwap, uint256 _majorPool, uint256 _minorPool) 
+    {
+        if(_minorInMajor == 0) {
+            _minorInMajor = getInputToOutputPrice(_minorAmount, _minorPoolSize, _majorPoolSize); 
+        }
+        _majorSwap = (_majorAmount.sub(_minorInMajor)).mul(1003).div(2000);
```

Comment:
I think the correct coefficient here is `1000/1997` instead of `1003/2000`:

We are looking for `_majorSwap` such that `_majorAmount - _majorSwap =  _minorInMajor + _majorSwap*.997`. Solving for `_majorSwap` gives `_majorSwap = (_majorAmount - _minorInMajor)/(1+.997)`

Reply:
Yes you are right indeed. This is what I had and for some reason I decided to change it.